src/Promise/FulfilledPromise.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Http\Client\Utils\Promise;
+
+use Http\Client\Exception;
+use Http\Client\Promise;
+use Psr\Http\Message\ResponseInterface;
+
+/**
+ * A promise already fulfilled
+ *
+ * @author Joel Wurtz <jwurtz@jolicode.com>
+ */
+class FulfilledPromise implements Promise
```

Comment:
do we not need a simple default implementation in the utils anyways? or
is that totally adapter specific?